### PR TITLE
Use CentralPackageVersions and Versions.props to ease maintenance of dependency versions

### PR DESCRIPTION
**Pull request by bpulliam**
For status checks on the develop branch, please use TransportPackage-Foundation
(https://microsoft.visualstudio.com/ProjectReunion/_build?definitionId=57248)
and run the build against your PR branch with the default parameters.

For status checks on the main branch, please use microsoft.ProjectReunion
(https://dev.azure.com/ms/ProjectReunion/_build?definitionId=391&_a=summary)
and run the build against your PR branch with the default parameters.

### CODE REVIEW

File: Packages.props [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Scottj1s) [inline Packages.props:1]: This looks like a mistaken checkin? Doesn't appear to be imported by anything, and C# projects should handle it themselves.  

Instead, we should have a root folder Directory.Packages.Props, with content like this:

```
<Project>
  <ItemGroup>
    <PackageVersion Include="Microsoft.Windows.CSWinRT" Version="$(CppWinRTVersion)"/>
    etc...
  </ItemGroup>
</Project>
```

The root Directory.Build.Props ensures that eng\Versions.props defines the version properties above.

File: dev/DeploymentAgent/DeploymentAgent.vcxproj [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\..\packages\Microsoft.Windows.CppWinRT.2.0.220929.3\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\..\packages\Microsoft.Windows.CppWinRT.2.0.220929.3\build\native\Microsoft.Windows.CppWinRT.props')" />
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <Import Project="..\..\packages\Microsoft.Windows.CppWinRT.$(CppWinRTVersion)\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\..\packages\Microsoft.Windows.CppWinRT.$(CppWinRTVersion)\build\native\Microsoft.Windows.CppWinRT.props')" />
```
Comment (DrusTheAxe) [inline dev/DeploymentAgent/DeploymentAgent.vcxproj:4]: Would really love to see all these relative paths go away e.g. "$(RepoRoot)\packages\..."

Good for a future PR. No need to hold up this PR for it

**Review by DrusTheAxe — APPROVED** [review]
LGTM

**Pull request merged**